$OpenBSD: patch-fat-ppc_c,v 1.1 2021/01/10 18:00:22 jasper Exp $

Revert https://git.lysator.liu.se/nettle/nettle/-/commit/db467660a1cbbd2bc65b8c09fc93546a9f391ae4
This breaks !__linux__ as __GLIBC_PREREQ is tested for and then used right after.

Index: fat-ppc.c
--- fat-ppc.c.orig
+++ fat-ppc.c
@@ -42,11 +42,12 @@
 
 #if defined(_AIX)
 # include <sys/systemcfg.h>
-#elif defined(__linux__) && defined(__GLIBC__) && \
-  defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 16)
-# define USE_GETAUXVAL 1
-# include <asm/cputable.h>
-# include <sys/auxv.h>
+#elif defined(__linux__)
+# if defined(__GLIBC__) && defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 16)
+#  define USE_GETAUXVAL 1
+#  include <asm/cputable.h>
+#  include <sys/auxv.h>
+# endif
 #elif defined(__FreeBSD__)
 # include <machine/cpu.h>
 # ifdef PPC_FEATURE2_HAS_VEC_CRYPTO
@@ -118,7 +119,7 @@ get_ppc_features (struct ppc_features *features)
 #else
       unsigned long hwcap = 0;
       unsigned long hwcap2 = 0;
-# if USE_GETAUXVAL
+# if defined(__linux__) && USE_GETAUXVAL
       hwcap = getauxval(AT_HWCAP);
       hwcap2 = getauxval(AT_HWCAP2);
 # elif defined(__FreeBSD__)
